### PR TITLE
Build: update templates to v0.1.4

### PR DESCRIPTION
**Pull request by Michkov**
- Use consume cluster templates from redhat-appstudio/build-definitions
  directly
- Move version config into one file components/build/build-templates/kustomization.yaml

### CODE REVIEW

File: argo-cd-apps/overlays/development/repo-overlay.yaml [hunk 1]
```diff
@@ -4,41 +4,41 @@ metadata:
   name: build
 spec:
   source: # This will be replaced with a reference to your fork of this repo (see hack/patch-apps-for-dev.sh)
-    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
-    targetRevision: main
+    repoURL: https://github.com/jduimovich/infra-deployments.git
```
Comment (sbose78) [inline argo-cd-apps/overlays/development/repo-overlay.yaml:7]: missed reverting this, I think.

Comment (jduimovich) [reply argo-cd-apps/overlays/development/repo-overlay.yaml:7]: My fault, I checked out the PR into my fork but pushed which updated the PR and not my branch from the fork

Comment (jduimovich) [reply argo-cd-apps/overlays/development/repo-overlay.yaml:7]: re-ran upstream mode and pushed

**Comment by sbose78** [comment]
Is this good to merge ?

**Comment by Michkov** [comment]
Probably we can wait for v0.1.4 which will remove ClusterTasks completely.

**Comment by Michkov** [comment]
@sbose78 now ready for merge, updated to v0.1.4. Tested on newly installed cluster using `./hack/build/m2-builds.sh`.

**Pull request merged**